### ISSUE TITLE
Clippy Warning in FromStr

### ISSUE THREAD

**Issue by smessmer** [frozenlib/parse-display#3]
Using this enum:

```rust
#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, FromStr, Display)]
pub enum MyFlag {
    #[display("o1")]
    OptionOne,
    #[display("o2")]
    OptionTwo,
}
```

causes a clippy warning:

```
warning: trivial regex
   --> lib.rs:
    |
841 | #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, FromStr, Display)]
    |                                                   ^^^^^^^
    |
    = help: consider using `==` on `str`s
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#trivial_regex
    = note: this warning originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
```

The generated FromStr macro could probably generate an `#[allow(clippy::trivial_regex)]` if this is intended, so that it can be used more easily in projects that enforce a clean clippy output.

**Comment by frozenlib** [comment]
Thanks for posting the issue.

This issue fix in 07c5b3772997a938749b46dac4155f5675f12827.

**Comment by smessmer** [comment]
Awesome, thanks for creating this library :)
